File: 029. Monotonic Array/MonotonicArray2.java
import java.util.*;

class Program {
    public static boolean isMonotonic(int[] array) {
        var isNonDecreasing = true;
        var isNonIncreasing = true;
        for (int i = 1; i < array.length; i++) {
            if (array[i] < array[i - 1]) {
                isNonDecreasing = false;
            }
            if (array[i] > array[i - 1]) {
                isNonIncreasing = false;
            }
        }
        return isNonDecreasing || isNonIncreasing;
    }
}

File: 029. Monotonic Array/MonotonicArray2.kt
package com.algoexpert.program

fun isMonotonic(array: List<Int>): Boolean {
    var isNonDecreasing = true
    var isNonIncreasing = true
    for (i in 1 until array.size) {
        if (array[i] < array[i - 1]) isNonDecreasing = false
        if (array[i] > array[i - 1]) isNonIncreasing = false
    }
    return isNonIncreasing || isNonDecreasing
}
